### ISSUE TITLE
plotting not correct if second y axis with large value

### ISSUE THREAD

**Issue by freevanx** [bokeh/bokeh#7169]
# READ AND FOLLOW THESE INSTRUCTIONS CAREFULLY

*ISSUES THAT DO NOT CONTAIN NECESSARY INFORMATION MAY BE CLOSED, IMMEDIATELY*

The issue tracker is NOT the place for general support. For questions and
technical assistance, come ask the [Bokeh mailing list](https://groups.google.com/a/continuum.io/forum/#!forum/bokeh) or join the chat on [Gitter](https://gitter.im/bokeh/bokeh). For feature requests, please provide a detailed description or proposal of the new capability or behavior.

For defects or deficiencies, please provide ALL OF THE FOLLOWING:

#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
OS: MacOS 10.12.6

python2 -V
Python 2.7.12

alembic (0.9.4)
appdirs (1.4.3)
appnope (0.1.0)
asn1crypto (0.22.0)
astroid (1.5.3)
attrs (17.1.0)
Automat (0.6.0)
backports-abc (0.5)
backports.functools-lru-cache (1.4)
backports.shutil-get-terminal-size (1.0.0)
backports.weakref (1.0.post1)
bcolz (0.12.1)
beautifulsoup4 (4.6.0)
bitarray (0.8.1)
bkcharts (0.2)
bleach (1.5.0)
bokeh (0.12.6)
Bottleneck (1.2.1)
certifi (2017.7.27.1)
cffi (1.10.0)
chardet (3.0.4)
click (6.7)
conda (4.3.16)
configparser (3.5.0)
constantly (15.1.0)
contextlib2 (0.5.5)
cryptography (1.8.1)
cssselect (1.0.1)
cycler (0.10.0)
cyordereddict (1.0.0)
Cython (0.26)
decorator (4.1.2)
empy (3.3.2)
empyrical (0.3.0)
entrypoints (0.2.3)
enum34 (1.1.6)
et-xmlfile (1.0.1)
filterpy (0.1.5)
Flask (0.12.2)
Flask-SocketIO (2.8.6)
funcsigs (1.0.2)
functools32 (3.2.3.post2)
futures (3.1.1)
hide-code (0.5.0)
holoviews (1.9.0)
html5lib (0.9999999)
idna (2.6)
incremental (16.10.1)
int-date (0.1.8)
intervaltree (2.1.0)
ipaddress (1.0.18)
ipykernel (4.6.1)
ipython (5.5.0)
ipython-genutils (0.2.0)
ipywidgets (7.0.1)
isort (4.2.15)
itsdangerous (0.24)
jdcal (1.3)
Jinja2 (2.9.6)
jsonschema (2.6.0)
jupyter (1.0.0)
jupyter-client (5.1.0)
jupyter-console (5.2.0)
jupyter-core (4.3.0)
jupyter-wysiwyg (0.1.3)
Keras (2.0.9)
lazy-object-proxy (1.3.1)
Logbook (1.1.0)
lru-dict (1.1.6)
lxml (3.7.3)
Mako (1.0.7)
Markdown (2.6.9)
MarkupSafe (1.0)
marshmallow (2.13.6)
matplotlib (2.0.2)
mccabe (0.6.1)
mistune (0.7.4)
mock (2.0.0)
multipledispatch (0.4.9)
nbconvert (5.3.1)
nbformat (4.4.0)
networkx (1.11)
nose (1.3.7)
notebook (5.2.0)
numexpr (2.6.2)
numpy (1.13.3)
openpyxl (2.4.0)
ordereddict (1.1)
packaging (16.8)
pandas (0.20.3)
pandas-datareader (0.5.0)
pandocfilters (1.4.2)
param (1.5.1)
parsel (1.2.0)
pathlib2 (2.3.0)
patsy (0.4.1)
pbr (3.1.1)
pdfkit (0.6.1)
pep8 (1.7.0)
pexpect (4.2.1)
pickleshare (0.7.4)
pip (9.0.1)
prompt-toolkit (1.0.15)
protobuf (3.3.0)
ptyprocess (0.5.2)
py (1.4.34)
PyAlgoTrade (0.18)
pyasn1 (0.2.3)
pyasn1-modules (0.0.8)
pycosat (0.6.1)
pycparser (2.17)
PyDispatcher (2.0.5)
pyftdi (0.13.2)
Pygments (2.2.0)
pylint (1.7.4)
pyOpenSSL (17.0.0)
pyparsing (2.2.0)
pyserial (3.3)
pytest (3.2.3)
python-dateutil (2.6.1)
python-editor (1.0.3)
python-engineio (1.5.2)
python-socketio (1.7.4)
python-xbrl (1.1.1)
pytrends (4.3.0)
pytz (2017.2)
pyusb (1.0.0)
PyYAML (3.12)
pyzmq (16.0.2)
qtconsole (4.3.1)
queuelib (1.4.2)
redis (2.10.5)
requests (2.18.4)
requests-file (1.4.2)
requests-ftp (0.3.1)
ruamel.ordereddict (0.4.9)
ruamel.yaml (0.15.18)
scandir (1.6)
scikit-learn (0.18.2)
scipy (0.19.0)
Scrapy (1.4.0)
service-identity (16.0.0)
setuptools (36.6.0)
simplegeneric (0.8.1)
simplejson (3.11.1)
singledispatch (3.4.0.3)
six (1.11.0)
sortedcontainers (1.5.7)
SQLAlchemy (1.1.9)
statsmodels (0.8.0)
stockstats (0.2.0)
subprocess32 (3.2.7)
TA-Lib (0.4.10)
tables (3.4.2)
TBB (0.1)
tensorflow (1.3.0)
tensorflow-tensorboard (0.1.8)
terminado (0.6)
testpath (0.3.1)
toolz (0.8.2)
tornado (4.5.2)
traitlets (4.3.2)
tushare (0.7.7)
Twisted (17.1.0)
urllib3 (1.22)
virtualenv (15.1.0)
w3lib (1.17.0)
wcwidth (0.1.7)
webencodings (0.5.1)
Werkzeug (0.12.2)
wheel (0.29.0)
widgetsnbextension (3.0.3)
wrapt (1.10.11)
yahoo-finance (1.4.0)
youtube-dl (2017.10.20)
zipline (1.1.1)
zope.interface (4.4.1)
#### Description of expected behavior and the observed behavior
When plotting a secondary y axis and the y data have very large values, the first y plotting is not correct, it still use a max axis as 2nd y.

#### Complete, minimal, self-contained example code that reproduces the issue

```
# code goes here between backticks
from bokeh.io import output_notebook, show
from bokeh.plotting import figure
from bokeh.models import LinearAxis, Range1d
from bokeh.layouts import gridplot

from datetime import tzinfo, timedelta, datetime, date
import pandas as pd
import pandas_datareader.data as web

output_notebook()

end = datetime.now()
start = datetime.now() - timedelta(days=365)
df = web.DataReader("NVDA", 'yahoo', start, end)

# create a new plot with a datetime axis type
plot2 = figure(plot_width=800, plot_height=300, x_axis_type="datetime")

plot2.line(df.index, df['Close'].values, color='navy', line_width=1.5)
# Setting the second y axis range name and range
plot2.extra_y_ranges = {"Vol2": Range1d(start=0, end=max(df['Volume'].values))}
plot2.line(df.index, df['Volume'].values, color='blue', line_width=1.5, y_range_name="Vol2")
# Adding the second axis to the plot.  
plot2.add_layout(LinearAxis(y_range_name="Vol2"), 'right')

show(plot2)
```

#### Screenshots or screencasts of the bug in action
![image](https://user-images.githubusercontent.com/3090863/32414380-d24e5f4e-c1ea-11e7-822c-79a760aac090.png)

**Comment by bryevdv** [comment]
By default `DataRange` ranges (which are the default if you don't otherwise specify) use *all* renderers for auto-ranging. If you want to limit thigns to just one or a subset of renders, you have ot ask for that explicitly:
```
r = plot2.line(df.index, df['Close'].values, color='red', line_width=1.5)
plot2.y_range.renderers = [r]
```

Since this really a usage/support question, I will close and ask that further discussion happen on the mailing list or stack overflow.
